appcues/src/main/java/com/appcues/analytics/Analytics.kt

```diff
@@ -0,0 +1,188 @@
+package com.appcues.analytics
+
+import com.appcues.AppcuesCoroutineScope
+import com.appcues.analytics.AnalyticsEvent.ScreenView
+import com.appcues.analytics.AnalyticsIntent.Anonymous
+import com.appcues.analytics.AnalyticsIntent.Event
+import com.appcues.analytics.AnalyticsIntent.Identify
+import com.appcues.analytics.AnalyticsIntent.UpdateGroup
+import com.appcues.analytics.AnalyticsIntent.UpdateProfile
+import com.appcues.analytics.AnalyticsIntentQueue.IntentProcessor
+import com.appcues.data.model.Experience
+import com.appcues.data.model.ExperienceTrigger.Qualification
+import com.appcues.data.model.QualificationResult
+import com.appcues.monitor.ApplicationMonitor
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.launch
+import java.util.Date
+
+internal interface QualificationService {
+
+    // will merge and hit the backend to see if it qualifies, returning QualificationResult
+    suspend fun qualify(intents: List<AnalyticsIntent>): QualificationResult?
+}
+
+internal interface RenderingService {
+
+    // will attempt to show experiences by order
+    // clearCache is mostly used when we know the qualification is from a screen_view
+    suspend fun show(experiences: List<Experience>, clearCache: Boolean)
+
+    suspend fun reset()
+}
+
+internal interface SessionService {
+
+    // returns whether session is started or not
+    suspend fun isSessionStarted(): Boolean
+
+    // starts session (can start using incoming intent or existing stored information
+    suspend fun startSession(intent: AnalyticsIntent): Boolean
+
+    // decorates incoming intent and returns a new intent with session/auto props information
+    suspend fun decorateIntent(intent: AnalyticsIntent): AnalyticsIntent
+
+    suspend fun reset()
+}
+
+// new model for Intents available for Analytics
+internal sealed class AnalyticsIntent(open val isInternal: Boolean, val timestamp: Date = Date()) {
```

Comment:
i think this is somewhat similar to the usage in `TrackingUpdate` on the iOS side, fyi https://github.com/appcues/appcues-ios-sdk/blob/main/Sources/AppcuesKit/Data/Analytics/TrackingUpdate.swift